bitnami/postgresql/templates/secrets.yaml

```
@@ -1,19 +1,49 @@
 {{- if (include "postgresql.createSecret" .) }}
+
+{{- $secretName := include "postgresql.fullname" . }}
+{{- $postgresqlPostgresPassword := "" }}
+{{- $postgresqlPassword := "" }}
+{{- $postgresqlReplicationPassword := "" }}
+
+{{- $secret := (lookup "v1" "Secret" .Release.Namespace $secretName) }}
+{{- if $secret }}
+  {{- if index $secret.data "postgresql-postgres-password" }}
+  {{- $postgresqlPostgresPassword = index $secret.data "postgresql-postgres-password" }}
+  {{- end -}}
+  {{- if index $secret.data "postgresql-password" }}
+  {{- $postgresqlPassword = index $secret.data "postgresql-password" }}
+  {{- end -}}
+  {{- if index $secret.data "postgresql-repliaction-password" }}
+  {{- $postgresqlReplicationPassword = index $secret.data "postgresql-replication-password" }}
+  {{- end -}}
+{{- end -}}
+
+{{- if not $postgresqlPostgresPassword }}
+{{- $postgresqlPostgresPassword = include "postgresql.postgres.password" . | b64enc | quote }}
+{{- end -}}
+{{- if not $postgresqlPassword }}
+{{- $postgresqlPassword = include "postgresql.password" . | b64enc | quote }}
+{{- end -}}
+{{- if not $postgresqlReplicationPassword }}
+{{- $postgresqlReplicationPassword = include "postgresql.replication.password" . | b64enc | quote }}
+{{- end -}}
+
+
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ template "postgresql.fullname" . }}
+  name: {{ $secretName }}
   labels:
   {{- include "common.labels.standard" . | nindent 4 }}
   {{- if .Values.commonAnnotations }}
   annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
   {{- end }}
 type: Opaque
 data:
-  postgresql-postgres-password: {{ include "postgresql.postgres.password" . | b64enc | quote }}
-  postgresql-password: {{ include "postgresql.password" . | b64enc | quote }}
+  postgresql-postgres-password: {{ $postgresqlPostgresPassword }}
```

```suggestion
  postgresql-postgres-password: {{ include "postgresql.postgres.password" . | b64enc | quote }}
```

I would leave the helper's call and modify said helper including the new code. It could be something similar to:

```
{{- $secret := (lookup "v1" "Secret" .Release.Namespace (include "postgresql.fullname" .)) }}
{{- define "postgresql.postgres.password" -}}
{{- if and $secret (index $secret.data "postgresql-postgres-password") }}
    {{- index $secret.data "postgresql-postgres-password" | b64dec }}
{{- else -}}
    {{- if .Values.global.postgresql.postgresqlPostgresPassword }}
        {{- .Values.global.postgresql.postgresqlPostgresPassword -}}
    {{- else if .Values.postgresqlPostgresPassword -}}
        {{- .Values.postgresqlPostgresPassword -}}
    {{- else -}}
        {{- randAlphaNum 10 -}}
    {{- end -}}
{{- end -}}
{{- end -}}
```